kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/encryption/EnvelopeEncryptionFilterIT.java

```diff
@@ -248,6 +257,78 @@ void produceAndConsumeEncryptedAndPlainTopicsAtSameTime(KafkaCluster cluster, To
         }
     }
 
+    /**
+     * Test that ensures that the record offsets returned by the broker are faithfully relayed to the client.
+     * @param cluster underlying kafka cluster
+     * @param compactedTopic topic configured for compaction.
+     * @param directConsumer consumer connected directly to the underlying kafka cluster
+     * @param testKmsFacade kms facade
+     * @throws Exception exception
+     */
+    @TestTemplate
+    @SuppressWarnings("java:S2925")
+    void offsetFidelity(@BrokerConfig(name = "log.cleaner.backoff.ms", value = "50") KafkaCluster cluster,
+                        @TopicConfig(name = "segment.ms", value = "125") @TopicConfig(name = "cleanup.policy", value = "compact") Topic compactedTopic,
+                        Consumer<String, String> directConsumer,
+                        TestKmsFacade<?, ?, ?> testKmsFacade)
+            throws Exception {
+        var testKekManager = testKmsFacade.getTestKekManager();
+        testKekManager.generateKek(compactedTopic.name());
+
+        var builder = proxy(cluster);
+
+        builder.addToFilters(buildEncryptionFilterDefinition(testKmsFacade));
+
+        try (var tester = kroxyliciousTester(builder);
+                var proxyProducer = tester.producer();
+                var proxyConsumer = tester.consumer()) {
+
+            proxyProducer.send(new ProducerRecord<>(compactedTopic.name(), "a", "a1"));
+            // Send two messages for key "b", the first will be eligible for compaction
+            proxyProducer.send(new ProducerRecord<>(compactedTopic.name(), "b", "b1"));
+            proxyProducer.send(new ProducerRecord<>(compactedTopic.name(), "b", "b2")).get(5, TimeUnit.SECONDS);
+
+            // Sleep for segment.ms so that the broker will begin a new segment when the next produce is received.
+            // The records in the first segment will become eligible for compaction.
+            Thread.sleep(125);
+            proxyProducer.send(new ProducerRecord<>(compactedTopic.name(), "c", "c1")).get(5, TimeUnit.SECONDS);
+
+            // Wait until the topic compaction has coalesced the two b records into one.
+            // This will result in a gap in the offsets.
+            var directlyReadRecords = await().atMost(Duration.ofSeconds(30))
+                    .pollDelay(Duration.ofSeconds(1))
+                    .until(() -> consumeAll(compactedTopic.name(), 0, directConsumer).map(EnvelopeEncryptionFilterIT::stringifyRecordKeyOffset).toList(),
+                            Matchers.contains("a:0", "b:2", "c:3"));
```

Comment:
The contains confuses me as the produced records are `a1, b1, b2, c1` but the assertion only checks for `0, 2, 3`....

OH wait no the assertion is checking the offset not the value! Ok. that is actually demonstrating what we want but certainly surprised me on first pass.  

It does lead me to wonder if we should be including the value as well to prove we didn't do something screwy and some how produce `b1` at offset `2` (I can't think of a realistic way for us to do that but want to test the idea)

Reply:
If I were to have the stringification of record include the _value_, the resulting string would contain the encrypted bytes.  So I'd no longer be able to make the assertion between the value obtained from the proxy and the value obtained directly.

Reply:
Yeah. That would rule out comparisons between the clusters. 

My two thoughts are:
1. Could we change the values to look more distinct from the offsets. `value_AA`  `value_BA` `value_BB` ... ?
2. Do we need an additional test to check the values at given offsets?  

I'm happy for 1 to be addressed without another review cycle and 2 happen in a separate PR if you feel it adds value.